internal/audit/local/badgerdb.go

```diff
@@ -122,37 +102,48 @@ func NewLog(conf *Conf, decisionFilter audit.DecisionLogEntryFilter) (*Log, erro
 	return l, nil
 }
 
-func (l *Log) RegisterCallback(fn func()) {
+func (l *Log) RegisterCallback(fn func(chan<- struct{})) {
 	l.callbackFn = fn
 }
 
 func (l *Log) batchWriter(maxBatchSize int, flushInterval time.Duration) {
 	batch := newBatcher(l.Db, maxBatchSize)
 	logger := l.logger.With(zap.String("component", "batcher"))
 
+	ticker := time.NewTicker(flushInterval)
+	defer ticker.Stop()
+
+	awaitCallbackFn := func() {
+		if l.callbackFn != nil {
+			ch := make(chan struct{})
+			go l.callbackFn(ch)
+			<-ch
```

Comment:
Since we are always waiting on `waitCh`, can't we just simplify this to just calling the function without a goroutine? 🤔 

```suggestion
			l.callbackFn()
```

Reply:
As it stands, if the server issues a backoff, the callback [implicitly retries in a separate goroutine](https://github.com/cerbos/cerbos/blob/9445e81c786fde9f3c6220420af7ab203dfa1dbd/internal/audit/hub/hub.go#L173), so we can't guarantee that it'll happen synchronously.

The alternative was that we could block for the duration of the retry cycle, but this would have the knock-on effect that the local storage flushes would be blocked also.

Reply:
Because of the `return` in line 174, nothing gets published to the `waitCh` until one of the retries succeeds, right? Isn't that the same as having a blocking a function call?